### PR TITLE
Fix bug in nav2_behavior_tree/bt_action_node

### PR DESCRIPTION
**Pull request by CihatAltiparmak**
---

## Basic Info

| Info | Please fill out this column |
| ------ | ----------- |
| Ticket(s) this addresses   | #3848 |
| Primary OS tested on | Ubuntu 22.04.3 LTS (Jammy Jellyfish) |
| Robotic platform tested on | - |

---

## Description of contribution in a few bullet points

* Fixed the bug in halt function inside
  nav2_behavior_tree/plugin/action/bt_action_node.hpp

* Added new case to nav2_behavior_tree/plugin/action/bt_action_node.hpp
  for testing the scenario to cancel

* Refactored existing cases in
  nav2_behavior_tree/plugin/action/bt_action_node.hpp

---
## Future work that may be required in bullet points

Test this on simulation platform.

---

#### For Maintainers: 
- [ ] Check that any new parameters added are updated in navigation.ros.org
- [ ] Check that any significant change is added to the migration guide
- [ ] Check that any new features **OR** changes to existing behaviors are reflected in the tuning guide
- [ ] Check that any new functions have Doxygen added
- [ ] Check that any new features have test coverage
- [ ] Check that any new plugins is added to the plugins page
- [ ] If BT Node, Additionally: add to BT's XML index of nodes for groot, BT package's readme table, and BT library lists

### CODE REVIEW

File: nav2_behavior_tree/include/nav2_behavior_tree/bt_action_node.hpp [hunk 1]
```diff
@@ -310,6 +311,16 @@ class BtActionNode : public BT::ActionNodeBase
           node_->get_logger(),
           "Failed to cancel action server for %s", action_name_.c_str());
       }
+
+      if (callback_group_executor_.spin_until_future_complete(future_result) !=
```
Comment (SteveMacenski) [inline nav2_behavior_tree/include/nav2_behavior_tree/bt_action_node.hpp:315]: Use the `server_timeout_` as a timeout so its not infinitely blocking

File: nav2_behavior_tree/include/nav2_behavior_tree/bt_action_node.hpp [hunk 2]
```diff
@@ -310,6 +311,16 @@ class BtActionNode : public BT::ActionNodeBase
...
+        rclcpp::FutureReturnCode::SUCCESS)
+      {
+        RCLCPP_ERROR(
+          node_->get_logger(),
+          "Failed to get result call failed :( for [%s]", action_name_.c_str());
```
Comment (SteveMacenski) [inline nav2_behavior_tree/include/nav2_behavior_tree/bt_action_node.hpp:320]:
```suggestion
          "Failed to get result for %s in node halt!", action_name_.c_str());
```

**Review by SteveMacenski — CHANGES_REQUESTED** [review]
Overall LGTM with a couple of details.

File: nav2_behavior_tree/include/nav2_behavior_tree/bt_action_node.hpp (hunk 2 above)
Comment (CihatAltiparmak) [reply nav2_behavior_tree/include/nav2_behavior_tree/bt_action_node.hpp]: Done it in new commit

**Review by SteveMacenski — APPROVED** [review]

**Comment by SteveMacenski** [comment]
Thanks!

**Pull request merged**